onnx/numpy_helper.py

```diff
@@ -272,6 +272,70 @@ def from_dict(dict, name=None):  # type: (Dict[Any, Any], Optional[Text]) -> Map
     return map
 
 
+def to_optional(optional):  # type: (OptionalProto) -> Optional[Any]
+    """Converts an optional def to a Python optional.
+
+    Inputs:
+        optional: an OptionalProto object.
+    Returns:
+        opt: the converted optional.
+    """
+    opt = None  # type: Optional[Any]
+    elem_type = optional.elem_type
+    value_field = mapping.OPTIONAL_ELEMENT_TYPE_TO_FIELD[elem_type]
+    value = getattr(optional, value_field)
+    # TODO: create a map and replace conditional branches
+    if elem_type == OptionalProto.TENSOR or elem_type == OptionalProto.SPARSE_TENSOR:
+        opt = to_array(value)
+    elif elem_type == OptionalProto.SEQUENCE:
+        opt = to_list(value)
+    elif elem_type == OptionalProto.MAP:
+        opt = to_dict(value)
+    else:
+        raise TypeError("The element type in the input optional is not supported.")
+    return opt
+
+
+def from_optional(opt, name=None, dtype=None):  # type: (Optional[Any], Optional[Text], Optional[int]) -> OptionalProto
+    """Converts an optional value into a Optional def.
+
+    Inputs:
+        opt: a Python optional
+        name: (optional) the name of the optional.
+        dtype: (optional) type of element in the input, used for specifying
+                          optional values when converting empty none.
+    Returns:
+        optional: the converted optional def.
+    """
+    # TODO: create a map and replace conditional branches
+    optional = OptionalProto()
+    if name:
+        optional.name = name
+
+    if dtype:
+        elem_type = dtype
```

Comment:
add some error checking around this? as in whether dtype falls within the expected values